src/janitor.c

```diff
@@ -485,7 +485,8 @@ static void check_pool_size(PgPool *pool)
 	    cur < pool->db->pool_size &&
 	    cf_pause_mode == P_NONE &&
 	    cf_reboot == 0 &&
-	    pool_client_count(pool) > 0)
+		(cf_min_pool_size_requires_clients == 0 || (cf_min_pool_size_requires_clients == 1
+				&& pool_client_count(pool) > 0)))
```

Comment:
This seems a simpler way of doing this, since we only support 0 and 1 for `min_pool_size_requires_clients`
```suggestion
		(cf_min_pool_size_requires_clients == 0 || pool_client_count(pool) > 0))
```